print() is a function in Python 3

flake8 testing of https://github.com/google/compare_gan on Python 3.6.3

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./compare_gan/src/datasets.py:86:16: F821 undefined name 'xrange'
      for i in xrange(range_start, range_end)
               ^
./compare_gan/src/gan_lib.py:198:38: E999 SyntaxError: invalid syntax
        print " [*] Training started!"
                                     ^
./compare_gan/src/image_similarity.py:296:14: F821 undefined name 'xrange'
    for k in xrange(len(power_factors)):
             ^
./compare_gan/src/simple_task_pb2.py:46:46: F821 undefined name 'unicode'
      has_default_value=False, default_value=unicode("", "utf-8"),
                                             ^
./compare_gan/src/simple_task_pb2.py:53:46: F821 undefined name 'unicode'
      has_default_value=False, default_value=unicode("", "utf-8"),
                                             ^
./compare_gan/src/gans/WGAN.py:42:35: E999 SyntaxError: invalid syntax
      print "Using Adam optimizer."
                                  ^
./compare_gan/src/gans/gans_with_penalty.py:73:35: E999 SyntaxError: invalid syntax
      print "Using Adam optimizer."
                                  ^
./compare_gan/src/gans/ops.py:141:35: E999 SyntaxError: invalid syntax
  print " [*] Spectral norm layers"
                                  ^
4     E999 SyntaxError: invalid syntax
4     F821 undefined name 'xrange'
8
```

--- a/compare_gan/src/datasets.py
+++ b/compare_gan/src/datasets.py
@@ -23,10 +23,7 @@ import os
 import numpy as np
 import tensorflow as tf
 
-try:
-    xrange          # Python 2
-except NameError:
-    xrange = range  # Python 3
+from builtins import range
 
 flags = tf.flags
 FLAGS = flags.FLAGS
@@ -88,7 +85,7 @@ def get_sharded_filenames(prefix, num_shards, range_start=0, range_end=None):
   return [
       os.path.join(FLAGS.dataset_root,
                    prefix + "-%05d-of-%05d" % (i, num_shards))
-      for i in xrange(range_start, range_end)
+      for i in range(range_start, range_end)
   ]
 
 

--- a/compare_gan/src/datasets.py
+++ b/compare_gan/src/datasets.py
@@ -23,6 +23,11 @@ import os
 import numpy as np
 import tensorflow as tf
 
+try:
+    xrange          # Python 2
+except NameError:
+    xrange = range  # Python 3
+
 flags = tf.flags
 FLAGS = flags.FLAGS
 

--- a/compare_gan/src/gan_lib.py
+++ b/compare_gan/src/gan_lib.py
@@ -14,6 +14,7 @@
 # limitations under the License.
 
 """Library used for training various flavors of GANs on various datasets."""
+from __future__ import print_function
 
 import collections
 import contextlib
@@ -195,6 +196,6 @@ def run_with_options(options, task_workdir, progress_reporter=None):
             result_dir=result_dir,
             gan_log_dir=gan_log_dir)
         gan.build_model()
-        print " [*] Training started!"
+        print(" [*] Training started!")
         gan.train(sess, progress_reporter)
-        print " [*] Training finished!"
+        print(" [*] Training finished!")

--- a/compare_gan/src/gans/WGAN.py
+++ b/compare_gan/src/gans/WGAN.py
@@ -15,6 +15,7 @@
 
 """Implementation of the WGAN algorithm (https://arxiv.org/abs/1701.07875)."""
 from __future__ import division
+from __future__ import print_function
 
 from compare_gan.src.gans.abstract_gan import AbstractGAN
 import tensorflow as tf
@@ -39,13 +40,13 @@ class WGAN(AbstractGAN):
 
   def get_optimizer(self, name_prefix):
     if self.optimizer == "adam":
-      print "Using Adam optimizer."
+      print("Using Adam optimizer.")
       return tf.train.AdamOptimizer(
           self.learning_rate,
           beta1=self.beta1,
           name=name_prefix + self.optimizer)
     elif self.optimizer == "rmsprop":
-      print "Using RMSProp optimizer."
+      print("Using RMSProp optimizer.")
       return tf.train.RMSPropOptimizer(
           self.learning_rate, name=name_prefix + self.optimizer)
     else:

--- a/compare_gan/src/gans/gans_with_penalty.py
+++ b/compare_gan/src/gans/gans_with_penalty.py
@@ -24,10 +24,7 @@ from compare_gan.src.gans.abstract_gan import AbstractGAN
 
 import tensorflow as tf
 
-try:
-    xrange          # Python 2
-except NameError:
-    xrange = range  # Python 3
+from builtins import range
 
 flags = tf.flags
 FLAGS = flags.FLAGS
@@ -166,7 +163,7 @@ class AbstractGANWithPenalty(AbstractGAN):
 
   def dragan_penalty(self, x, discriminator, is_training):
     """Returns the DRAGAN gradient penalty."""
-    _, var = tf.nn.moments(x, axes=list(xrange(len(x.get_shape()))))
+    _, var = tf.nn.moments(x, axes=list(range(len(x.get_shape()))))
     std = tf.sqrt(var)
     x_noisy = x + std * (tf.random_uniform(x.shape) - 0.5)
     x_noisy = tf.clip_by_value(x_noisy, 0.0, 1.0)

--- a/compare_gan/src/gans/gans_with_penalty.py
+++ b/compare_gan/src/gans/gans_with_penalty.py
@@ -15,6 +15,7 @@
 
 """Implementation of GANs (MMGAN, NSGAN, WGAN) with different regularizers."""
 from __future__ import division
+from __future__ import print_function
 
 from compare_gan.src.gans import consts
 from compare_gan.src.gans import dcgan_architecture
@@ -22,6 +23,11 @@ from compare_gan.src.gans import resnet_architecture
 from compare_gan.src.gans.abstract_gan import AbstractGAN
 
 import tensorflow as tf
+
+try:
+    xrange          # Python 2
+except NameError:
+    xrange = range  # Python 3
 
 flags = tf.flags
 FLAGS = flags.FLAGS
@@ -70,18 +76,18 @@ class AbstractGANWithPenalty(AbstractGAN):
 
   def get_optimizer(self, name_prefix):
     if self.optimizer == "adam":
-      print "Using Adam optimizer."
+      print("Using Adam optimizer.")
       return tf.train.AdamOptimizer(
           self.learning_rate,
           beta1=self.beta1,
           beta2=self.beta2,
           name=name_prefix + self.optimizer)
     elif self.optimizer == "rmsprop":
-      print "Using RMSProp optimizer."
+      print("Using RMSProp optimizer.")
       return tf.train.RMSPropOptimizer(
           self.learning_rate, name=name_prefix + self.optimizer)
     elif self.optimizer == "sgd":
-      print "Using GradientDescent optimizer."
+      print("Using GradientDescent optimizer.")
       return tf.train.GradientDescentOptimizer(self.learning_rate)
     else:
       raise ValueError("Unknown optimizer: %s" % self.optimizer)
@@ -187,7 +193,7 @@ class AbstractGANWithPenalty(AbstractGAN):
     t_vars = tf.trainable_variables()
     d_vars = [var for var in t_vars if "discriminator" in var.name]
     # Linear + conv2d and deconv2d layers.
-    print d_vars
+    print(d_vars)
     d_weights = [
         v for v in d_vars
         if ((v.name.endswith("/Matrix:0") and "fc" in v.name) or

--- a/compare_gan/src/gans/ops.py
+++ b/compare_gan/src/gans/ops.py
@@ -14,6 +14,7 @@
 # limitations under the License.
 
 from __future__ import division
+from __future__ import print_function
 
 import numpy as np
 import scipy.misc
@@ -138,12 +139,12 @@ def linear(input_,
 
 def spectral_norm_update_ops(var_list, weight_getter):
   update_ops = []
-  print " [*] Spectral norm layers"
+  print(" [*] Spectral norm layers")
   layer = 0
   for var in var_list:
     if weight_getter.match(var.name):
       layer += 1
-      print "     %d. %s" % (layer, var.name)
+      print("     %d. %s" % (layer, var.name))
       # Alternative solution here is keep spectral norm and original weight
       # matrix separately, and only normalize the weight matrix if needed.
       # But as spectral norm converges to 1.0 very quickly, it should be very

--- a/compare_gan/src/image_similarity.py
+++ b/compare_gan/src/image_similarity.py
@@ -39,10 +39,7 @@ from __future__ import print_function
 
 import tensorflow as tf
 
-try:
-    xrange          # Python 2
-except NameError:
-    xrange = range  # Python 3
+from builtins import range
 
 
 def VerifyCompatibleImageShapes(img1, img2):
@@ -298,7 +295,7 @@ def _MultiscaleSSIMHelper(
       return [tf.pad(x, padding, mode='SYMMETRIC') for x in images]
 
     mcs = []
-    for k in xrange(len(power_factors)):
+    for k in range(len(power_factors)):
       with tf.name_scope(None, 'Scale%d' % k, imgs):
         if k > 0:
           # Avg pool takes rank 4 tensors. Flatten leading dimensions.

--- a/compare_gan/src/image_similarity.py
+++ b/compare_gan/src/image_similarity.py
@@ -39,6 +39,11 @@ from __future__ import print_function
 
 import tensorflow as tf
 
+try:
+    xrange          # Python 2
+except NameError:
+    xrange = range  # Python 3
+
 
 def VerifyCompatibleImageShapes(img1, img2):
   """Checks if two image tensors are compatible for metric computation.

--- a/compare_gan/src/simple_task_pb2.py
+++ b/compare_gan/src/simple_task_pb2.py
@@ -43,14 +43,14 @@ _TASKDIMENSION = _descriptor.Descriptor(
     _descriptor.FieldDescriptor(
       name='parameter', full_name='compare_gan.TaskDimension.parameter', index=0,
       number=1, type=9, cpp_type=9, label=1,
-      has_default_value=False, default_value=unicode("", "utf-8"),
+      has_default_value=False, default_value=u"",
       message_type=None, enum_type=None, containing_type=None,
       is_extension=False, extension_scope=None,
       options=None),
     _descriptor.FieldDescriptor(
       name='string_value', full_name='compare_gan.TaskDimension.string_value', index=1,
       number=2, type=9, cpp_type=9, label=1,
-      has_default_value=False, default_value=unicode("", "utf-8"),
+      has_default_value=False, default_value=u"",
       message_type=None, enum_type=None, containing_type=None,
       is_extension=False, extension_scope=None,
       options=None),

